float8_experimental/float8_linear_utils.py

```diff
@@ -114,6 +115,9 @@ def post_order_traversal(
             setattr(parent_module, module_name, module_cls.from_float(module, emulate))
 
     post_order_traversal(root_module, "", None)
+    # Without this explicit `del`, this set only gets deleted upon an explicit
+    # garbage collection (not from when its refcount hits zero)
+    del visited_modules
```

Comment:
interesting, is there a hash or smaller uuid we can use to mark as visited, I would imagine that this could cause a spike during swapping, that would likely be less than start of backwards but idk models can be weird

Reply:
I think outside of FSDP, there is no memory issue. The new `Float8Linear.weight` shares a reference to the existing `nn.Linear.weight`, so there is no duplication.

The issue was that for FSDP, we expected the original `Float8Linear.weight` / `nn.Linear.weight` to get deleted quickly after sharding and replacing `Float8Linear.weight` with a sharded weight.

Reply:
Ahhhh okay